src/host/posix/dnssd.cpp

```diff
@@ -113,62 +118,80 @@ extern "C" void otPlatDnssdUnregisterKey(otInstance                 *aInstance,
 
 extern "C" void otPlatDnssdStartBrowser(otInstance *aInstance, const otPlatDnssdBrowser *aBrowser)
 {
-    OTBR_UNUSED_VARIABLE(aInstance);
-    OTBR_UNUSED_VARIABLE(aBrowser);
+    otbr::DnssdPlatform::Get().StartServiceBrowser(
+        *aBrowser, MakeDiscoveryCallbackIdentifier(aBrowser->mCallback),
+        [aInstance, aBrowser](const otbr::DnssdPlatform::BrowseResult &aResult) {
+            aBrowser->mCallback(aInstance, &aResult);
```

Comment:
I haven't looked into the OT implementation in details. I wonder if `aBrowser` is guaranteed to be available in this lambda? Same for the `aResolver` in the following code.